Add a distro version check pattern which allows a character in distro version
string, such as 10.20.21a.0.

Upstream-Status: Inappropriate [distro specific]

Signed-off-by: Kai Kang <kai.kang@windriver.com>

diff --git a/productmd/treeinfo.py b/productmd/treeinfo.py
index 58429f9..6d9fb80 100644
--- a/productmd/treeinfo.py
+++ b/productmd/treeinfo.py
@@ -171,7 +171,8 @@ class BaseProduct(productmd.common.MetadataBase):
     def _validate_version(self):
         self._assert_type("version", list(six.string_types))
         if re.match(r'^\d', self.version):
-            self._assert_matches_re("version", [r"^\d+(\.\d+)*$"])
+            patterns = [r"^\d+(\.\d+)*$", r"^\d+\.\d+\.\d+[a-zA-Z]?\.\d+"]
+            self._assert_matches_re("version", patterns)
 
     def _validate_short(self):
         self._assert_type("short", list(six.string_types))
